### DISCUSSION TITLE
Open sourcing the GUI framework

### DISCUSSION THREAD

**Discussion by srounce** [warpdotdev/Warp#430] [Company announcements]
Are there any plans to open source any of the layers of the GUI framework described in [this article](https://blog.warp.dev/how-warp-works/) any time soon?

**Comment by alokedesai** [comment]
Thank you for filing! This is absolutely something we plan on open sourcing once it's more mature, but we don't have a concrete timeline for when that will be just yet. Our main goal is to make it cross platform first, since it only targets MacOS right now.

**Reply by bglw** [reply]
Do you have an idea of what the license will be for the GUI framework?

**Reply by jasonwun** [reply]
It is been 4 years and now warp is available in macos + linux + windows. Is it on the team's radar now?

**Reply by julibuilds** [reply]
I think a lot of people would love to have this

**Comment by alokedesai** [comment]
Hey everyone! We’ve spent the past few years hard at work building Warp and improving our UI framework. Today, we open sourced the entire Warp client, including our UI framework. The UI framework compiles to Mac, Linux, Windows, and the Web (via WASM) is available with a permissive MIT license so others can build on top of it. 


Thank you so much to all of you for pushing us early on to do this. More in the pinned announcement [discussion](https://github.com/warpdotdev/warp/discussions/9240) and the [blog](https://www.warp.dev/blog/warp-is-now-open-source).

**Reply by srounce** [reply]
Yes I saw the post and remembered this issue, thanks for remembering :+1:
